lazer/sdk/js/src/socket/websocket-pool.ts

```diff
@@ -8,6 +8,77 @@ import type { Request, Response } from "../protocol.js";
 import type { ResilientWebSocketConfig } from "./resilient-websocket.js";
 import { ResilientWebSocket } from "./resilient-websocket.js";
 
+/**
+ * Detects if the code is running in a regular DOM or Web Worker context.
+ * @returns true if running in a DOM or Web Worker context, false if running in Node.js
+ */
+function isBrowser(): boolean {
+  try {
+    // Check for browser's window object (DOM context)
+    if (typeof globalThis !== "undefined") {
+      const global = globalThis as any;
+      if (global.window && global.window.document) {
+        return true;
+      }
+
+      // Check for Web Worker context (has importScripts but no window)
+      if (typeof global.importScripts === "function" && !global.window) {
+        return true;
+      }
+    }
+
+    // Node.js environment
+    return false;
+  } catch {
+    // If any error occurs, assume Node.js environment
+    return false;
+  }
+}
+
+/**
+ * Adds authentication to either the URL as a query parameter or as an Authorization header.
+ * 
+ * Browsers don't support custom headers for WebSocket connections, so if a browser is detected,
+ * the token is added as a query parameter instead. Else, the token is added as an Authorization header.
+ * 
+ * @param url - The WebSocket URL
+ * @param token - The authentication token
+ * @param wsOptions - Existing WebSocket options
+ * @returns Object containing the modified endpoint and wsOptions
+ */
+function addAuthentication(
+  url: string,
+  token: string,
+  wsOptions: any = {}
```

Comment:
In general you shouldn't ever use `any`; here you should type this correctly to match the types that are valid as options here.  If you must, you can use `unknown`, but I don't think that makes sense in this case.